### PR TITLE
feat(api,hooks): add and trigger workflow hooks

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19160.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19161.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19162.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Review by yesnault — APPROVED** [review]

File: engine/api/migrate/migrate_project_repositories.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"context"
+	"strings"
+
+	"github.com/go-gorp/gorp"
+
+	"github.com/ovh/cds/engine/api/repository"
+	"github.com/ovh/cds/sdk"
+)
+
+func MigrateProjectRepsositories(ctx context.Context, db *gorp.DbMap) error {
```
Comment (richardlt) [inline engine/api/migrate/migrate_project_repositories.go:13]:
```suggestion
func MigrateProjectRepositories(ctx context.Context, db *gorp.DbMap) error {
```

**Review by richardlt — APPROVED** [review]

**Review by fsamin — APPROVED** [review]

File: engine/api/migrate/migrate_project_repositories.go (hunk 1 above)
Comment (sguiheux) [reply engine/api/migrate/migrate_project_repositories.go]: fixed

**Pull request merged**